haskellPackages: stackage LTS 24.16 -> LTS 24.19

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `staging`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `staging` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #452254. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

sternenseemann committed:
haskellPackages: _ prefix attribute names that would need quoting

In order not to break downstream expressions that (manually) introduced
dependencies on now renamed attributes, add aliases (with proper
warnings about the renames) for all attributes that were previously part
of Nixpkgs, but have been renamed with this change.

sternenseemann committed:
haskell.packages.ghc94.haskell-language-server: fix ref to config

wolfgangwalther committed:
haskell.packages.ghc96.haskell-language-server: fix build

This dependency is only needed for the "plugin-tutorial" executable.
This is only built with GHC 9.6 and 9.8, but the expression is created
with GHC 9.10. Thus add the missing dependency.

It would be better to just disable that tutorial, but I didn't find a
quick way of doing that - toggling the related flag did not work.

This only fixed the build for GHC 9.6 for me, GHC 9.8 failed with a
different error right after.

(cherry picked from commit a36931ccf2c4a5d146c5617708ad199276e2547a)

wolfgangwalther committed:
haskell.packages.ghc98.haskell-language-server: fix build

It was complaining about different versions of some regex dependency in
the closure, the differing one introduced by hls-plugin-api.

GHC 9.6 didn't fail that way, but I figured such a change can't hurt
either, for consistency.

(cherry picked from commit 7f8ab22df9c6e9ffb088e3e6b2edd387df9a9b10)

sternenseemann committed:
haskellPackages.duckdb-ffi: mark as broken

I've decided to wait on upstream feedback on
https://github.com/Tritlo/duckdb-haskell/issues/2 before trying to get
it to build again.

(cherry picked from commit 3c9a1a34c8e0229fc542c43497cb4423778b3c56)

alexfmpe committed:
haskellPackages.proto3-suite: fix build

(cherry picked from commit 02b355acd2a813b00ddfb496ab6c872392fbdc81)

wolfgangwalther committed:
Merge commit 'cef1398c495876ef5e5b02fb4f6291ceb3f6f5b5' into haskell-updates

nixpkgs-ci[bot] committed:
Merge cc9d3a57daf6d1b3542326db274d4fa50ea81836 into haskell-updates

nixpkgs-ci[bot] committed:
Merge de2bd218354f824294170423eb7808f8a1a611d8 into haskell-updates

wolfgangwalther committed:
postgrest: 14.0 -> 14.1

Release Notes:
https://github.com/PostgREST/postgrest/releases/tag/v14.1

sternenseemann committed:
haskellPackages.pdftotext: move jailbreak into overrideCabal

(cherry picked from commit 18e4ed738312e03f66e5b9b0b458beeed0e5a9a1)

sternenseemann committed:
haskellPackages: stackage LTS 24.16 -> LTS 24.19

all-cabal-hashes: 2025-10-22T21:20:29Z -> 2025-11-11T08:13:55Z

(generated by maintainers/scripts/haskell/update-package-set.sh)

haskellPackages.cabal2nix-unstable: 2025-10-31 -> 2025-11-11

This eliminates the need for 2f1f72d5a2ea8d96f5ba200567815cb6d43d79f3.

sternenseemann committed:
haskellPackages.pdftotext: remove redundant poppler input

Automatically provided since 5a80601f9795a42d12c6358a3fa10c025df79046.

sternenseemann committed:
haskellPackages.crucible*: adjust to package updates

- Some patches have been released properly.
- what4 1.7.1.0 has been released.

sternenseemann committed:
haskellPackages.gi-gtk-hs: drop obsolete override

sternenseemann committed:
Merge commit 61630d4ab5d87936 into haskell-updates

mpscholten committed:
haskellPackages.text-icu: disable homebrew flag on darwin

By default text-icu sets homebrew=true. This can cause a nix build
of text-icu to accidentally link the homebrew icu. It can also
cause "doesn't work on my machine" issues when an old homebrew version
with icu is installed on a mac and is accidentally picked up.

wolfgangwalther committed:
haskellPackages.text-icu: disable homebrew flag (#462106)

wolfgangwalther committed:
Merge commit 'c840f3de26e798e19e0e6ed7bf8decf9cfb34d2f' into haskell-updates

nixpkgs-ci[bot] committed:
Merge da72f8229bc9f1d9a64cfa54d1b35fd6f2bd0eb6 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 225bffcec8104e19a69efdf52c2395e3f00fd1c5 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 482eaefbd43e065787da7521c43bdf427c5d83b2 into haskell-updates

sternenseemann committed:
haskellPackages.cabal2nix-unstable: 2025-11-11 -> 2025-11-20

Also run Nix based test cases of the language-nix test suite.

nixpkgs-ci[bot] committed:
Merge baa40a5be0cd536fc753a74fe763a75476abb2a7 into haskell-updates

sternenseemann committed:
git-annex: update sha256 for 10.20251029

sternenseemann committed:
haskellPackages.proto3-wire: update list of required patches

1.4.4 incorporates most of the patches we need, but introduces a too
strict upper bound on doctest we need to lift.

sternenseemann committed:
top-level/release-haskell.nix: stop redundantly git-annex w/ 9.10

sternenseemann committed:
git-annex: remove compat code for GHC < 9.10

This was implemented when we were using 9.8 in order to ease the
transition to 9.10, but we don't really need to be able to build
git-annex on 9.8 anymore.

nixpkgs-ci[bot] committed:
Merge 64a5bece096e4d76a87b1b4c41e055253b95839b into haskell-updates

sternenseemann committed:
git-annex: remove compat code for GHC < 9.10 (#463791)

sternenseemann committed:
haskell.packages.ghc948.crypton-x509-store: pin to 1.6.11

nixpkgs-ci[bot] committed:
Merge 60d40a82c9c7775be1713df341885790d8065bcf into haskell-updates

nixpkgs-ci[bot] committed:
Merge dbff0491a2f9961ba647b5f022195891c2924254 into haskell-updates

wolfgangwalther committed:
haskell.packages.ghc948.crypton-x509-store: pin to 1.6.11 (#463970)

wolfgangwalther committed:
haskell.packages.ghc94.hlint: pin to 3.8

We use the same version for GHC 9.6 and GHC 9.8, so it's no additional
effort to pin here as well. This pin was lost when turning HLS into a
throw.

wolfgangwalther committed:
haskellPackages: stackage LTS 24.19 -> LTS 24.21

all-cabal-hashes: 2025-11-11T08:13:55Z -> 2025-11-24T11:39:51Z

(generated by maintainers/scripts/haskell/update-package-set.sh)

wolfgangwalther committed:
haskell.packages.ghc94.hlint: pin to 3.6.1

My previous commit pinned to the wrong version, the version used for 9.4
was indeed 9.6.1. This time I confirmed building locally as well...

wolfgangwalther commented:
Everything for this branch except the [`haskell.packages.ghc984.haskell-language-server` failure](https://hydra.nixos.org/build/314122209/nixlog/1) has fixes merged or in PRs already. The latter fails with:

```
Warning:
    This package indirectly depends on multiple versions of the same package. This is very likely to cause a compile failure.
      package stylish-haskell (stylish-haskell-0.14.6.0-4VhMq1uGahLJ7qA3wPHBy9) requires regex-tdfa-1.3.2.5-7dkScWcVpzn3V2HwE9STa9
      package hls-plugin-api (hls-plugin-api-2.12.0.0-5wixVw5qXZp1kc03B11Qwa) requires regex-tdfa-1.3.2.5-7dkScWcVpzn3V2HwE9STa9
      package ghcide (ghcide-2.12.0.0-CcuavIhfkAeJYLOeGg22WJ) requires regex-tdfa-1.3.2.5-7dkScWcVpzn3V2HwE9STa9
      package haskell-language-server (haskell-language-server-2.12.0.0) requires regex-tdfa-1.3.2.5-EDTc5eeC5AT3KJl322Zytc
      package haskell-language-server (haskell-language-server-2.12.0.0) requires regex-tdfa-1.3.2.5-EDTc5eeC5AT3KJl322Zytc
      package haskell-language-server (haskell-language-server-2.12.0.0) requires regex-tdfa-1.3.2.5-EDTc5eeC5AT3KJl322Zytc
      package haskell-language-server (haskell-language-server-2.12.0.0) requires regex-tdfa-1.3.2.5-EDTc5eeC5AT3KJl322Zytc
      package haskell-language-server (haskell-language-server-2.12.0.0) requires regex-tdfa-1.3.2.5-EDTc5eeC5AT3KJl322Zytc
      package haskell-language-server (haskell-language-server-2.12.0.0) requires regex-tdfa-1.3.2.5-EDTc5eeC5AT3KJl322Zytc
```

Of course, this fails only in hydra again, not locally for me.

`nix-diff` has this to say:

```
- /nix/store/v38ps9bf4w9489isagi2mz4dc2gf32jj-regex-tdfa-1.3.2.5.drv:{out}
+ /nix/store/dybfjbsig2limjcpbda48gv7wfga3bfj-regex-tdfa-1.3.2.5.drv:{out}
• The input derivation named `doctest-parallel-0.4` differs
  - /nix/store/z9qw7bhik77pw8kqlvavm7w59mzfi5wh-doctest-parallel-0.4.drv:{out}
  + /nix/store/azkbq5nb2080qlkbzva1ylj0cqip5hvg-doctest-parallel-0.4.drv:{out}
  • The set of input derivation names do not match:
      - Cabal-3.10.3.0
      + Cabal-3.14.2.0
  • The input derivation named `ghc-exactprint-1.8.0.0` differs
    - /nix/store/sp984kn6n561j0kg1laac2bypkg65q4v-ghc-exactprint-1.8.0.0.drv:{out}
    + /nix/store/kxg8vn6811qsmnhi08zc3ixv9nw6h86l-ghc-exactprint-1.8.0.0.drv:{out}
    • The set of input derivation names do not match:
        - Cabal-syntax-3.10.3.0
        + Cabal-syntax-3.14.2.0
```

I didn't go further, yet.

wolfgangwalther committed:
haskellPackages: run unbreak.nu

created by running maintainers/scripts/haskell/unbreak.nu

nixpkgs-ci[bot] committed:
Merge de3255d9ebab95d73cef7167991a8098083541a9 into haskell-updates

wolfgangwalther committed:
haskellPackages: stackage LTS 24.19 -> LTS 24.21 (#464592)

nixpkgs-ci[bot] committed:
Merge f087502faa7cd5fc268327cb7092e0383c65fa71 into haskell-updates

sternenseemann committed:
haskellPackages: _ prefix attribute names that would need quoting (#453366)

nixpkgs-ci[bot] committed:
Merge 45af851ce7b0c1a95d7d901bb7a9816c3cb2378f into haskell-updates

nixpkgs-ci[bot] committed:
Merge fb451ae6708d0cc2ee8378c241448e89d446cd52 into haskell-updates

nixpkgs-ci[bot] committed:
Merge bc0da6f230efdc7772bac7f1e6f925da75145728 into haskell-updates

sternenseemann committed:
git-annex: update sha256 for 10.20251114

sternenseemann committed:
hledger-check-fancyassertions: update sha256 for 1.50.3

nixpkgs-ci[bot] committed:
Merge 6a49303095abc094ee77dc243a9e351b642e8e75 into haskell-updates

wolfgangwalther committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by
maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1820566](https://hydra.nixos.org/eval/1820566) of nixpkgs
commit
[62a3860](https://github.com/NixOS/nixpkgs/commits/62a3860c3400eec2f9d236fbd6590e04102b15b2)
as of 2025-11-29 12:05 UTC*
from the haskell-updates jobset on hydra under
https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

wolfgangwalther committed:
haskellPackages._4Blocks: fix transitive broken

wolfgangwalther commented on pkgs/development/haskell-modules/configuration-hackage2nix/transitive-broken.yaml:
```diff
@@ -3905,3 +3918,4 @@ dont-distribute-packages:
  - zoovisitor
  - zuramaru
  - zwirn
+ - _4Blocks
```
Ah, I see now. This *did* cause problems, but it didn't cause hackage2nix to error out. It did however not match the package, so removed the `hydraPlatforms = none` line we want here. I reversed that manually, will fix the `type` thing manually for now, too.

wolfgangwalther committed:
haskellPackages._type: mark broken

The script didn't pick this up correctly, so manually fixing for now.

wolfgangwalther commented:
I opened a backport PR for this at #466256.
